src/layer/tile/TileLayer.WMS.js

```diff
@@ -23,48 +23,50 @@ import {Bounds} from '../../geometry/Bounds.js';
 // Instantiates a WMS tile layer object given a base URL of the WMS service and a WMS parameters/options object.
-export const TileLayerWMS = TileLayer.extend({
-
-	// @section
-	// @aka TileLayer.WMS options
-	// If any custom options not documented here are used, they will be sent to the
-	// WMS server as extra parameters in each request URL. This can be useful for
-	// [non-standard vendor WMS parameters](https://docs.geoserver.org/stable/en/user/services/wms/vendor.html).
-	defaultWmsParams: {
-		service: 'WMS',
-		request: 'GetMap',
-
-		// @option layers: String = ''
-		// **(required)** Comma-separated list of WMS layers to show.
-		layers: '',
-
-		// @option styles: String = ''
-		// Comma-separated list of WMS styles.
-		styles: '',
-
-		// @option format: String = 'image/jpeg'
-		// WMS image format (use `'image/png'` for layers with transparency).
-		format: 'image/jpeg',
-
-		// @option transparent: Boolean = false
-		// If `true`, the WMS service will return images with transparency.
-		transparent: false,
-
-		// @option version: String = '1.1.1'
-		// Version of the WMS service to use
-		version: '1.1.1'
-	},
-
-	options: {
-		// @option crs: CRS = null
-		// Coordinate Reference System to use for the WMS requests, defaults to
-		// map CRS. Don't change this if you're not sure what it means.
-		crs: null,
-
-		// @option uppercase: Boolean = false
-		// If `true`, WMS request parameter keys will be uppercase.
-		uppercase: false
-	},
+export class TileLayerWMS extends TileLayer {
+
+	static {
+		// @section
+		// @aka TileLayer.WMS options
+		// If any custom options not documented here are used, they will be sent to the
+		// WMS server as extra parameters in each request URL. This can be useful for
+		// [non-standard vendor WMS parameters](https://docs.geoserver.org/stable/en/user/services/wms/vendor.html).
+		this.prototype.defaultWmsParams = {
+			service: 'WMS',
+			request: 'GetMap',
+
+			// @option layers: String = ''
+			// **(required)** Comma-separated list of WMS layers to show.
+			layers: '',
+
+			// @option styles: String = ''
+			// Comma-separated list of WMS styles.
+			styles: '',
+
+			// @option format: String = 'image/jpeg'
+			// WMS image format (use `'image/png'` for layers with transparency).
+			format: 'image/jpeg',
+
+			// @option transparent: Boolean = false
+			// If `true`, the WMS service will return images with transparency.
+			transparent: false,
+
+			// @option version: String = '1.1.1'
+			// Version of the WMS service to use
+			version: '1.1.1'
+		};
+
+		this.setDefaultOptions({
+			// @option crs: CRS = null
+			// Coordinate Reference System to use for the WMS requests, defaults to
+			// map CRS. Don't change this if you're not sure what it means.
+			crs: null,
+
+			// @option uppercase: Boolean = false
+			// If `true`, WMS request parameter keys will be uppercase.
+			uppercase: false
+		});
+	}
 
 	initialize(url, options) {

```

Comment:
Eventually, we can migrate the remaining `initialize` to proper ES6 class `constructor`